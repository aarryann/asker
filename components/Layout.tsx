// components/Layout.tsx

import { NextPage } from 'next';
import Head from 'next/head';

import Header from './Header';
import NavBar from './NavBar';
import styles from './Layout.module.scss';
import navButtons from '../config/buttons';

type Props = {
  children: React.ReactNode;
};
const Layout: NextPage<Props> = ({ children }) => {
  const appTitle = '> WHATABYTE';

  return (
    <div className={styles.Layout}>
      <Head>
        <title>WHATABYTE</title>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta charSet="utf-8" />
      </Head>

      <Header appTitle={appTitle} />
      <div className={styles.Content}>{children}</div>
      <NavBar navButtons={navButtons} />
    </div>
  );
};

export default Layout;
