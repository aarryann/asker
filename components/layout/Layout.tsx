// components/Layout.tsx

import { NextPage } from 'next';
import Head from 'next/head';

import Header from './Header';
import NavBar from './NavBar';
import styles from './Layout.module.scss';
import navButtons from '../../config/buttons';

type Props = {
  children: React.ReactNode;
};
const Layout: NextPage<Props> = ({ children }) => {
  const appTitle = '> WHATABYTE';

  return (
    <div id="wrapper" className={styles.Layout}>
      <Head>
        <title>WHATABYTE</title>
      </Head>

      <Header appTitle={appTitle} />
      <div id="content" className={styles.Content}>
        {children}
      </div>
      <NavBar navButtons={navButtons} />
    </div>
  );
};

export default Layout;
