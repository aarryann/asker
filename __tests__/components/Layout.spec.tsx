import React from 'react';
import { shallow } from 'enzyme';
import Layout from '@components/layout/Layout';

describe('Layout', () => {
  let wrapper;
  const appTitle = 'Test';

  // eslint-disable-next-line
  beforeEach(() => (wrapper = shallow(<Layout>{appTitle}</Layout>)));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render children', () => {
    expect(wrapper.find('div#wrapper').length).toEqual(1);
    expect(wrapper.find('Header')).toHaveLength(1);
    expect(wrapper.find('div#content').length).toEqual(1);
    expect(wrapper.find('NavBar')).toHaveLength(1);
  });

  it('renders the value passed', () => {
    expect(wrapper.find('div#content').text()).toEqual(appTitle);
  });
});
