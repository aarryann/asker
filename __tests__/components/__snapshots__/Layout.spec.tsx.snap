// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<div
  id="wrapper"
>
  <Head>
    <title>
      WHATABYTE
    </title>
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
    <meta
      charSet="utf-8"
    />
  </Head>
  <Header
    appTitle="> WHATABYTE"
  />
  <div
    id="content"
  >
    Test
  </div>
  <NavBar
    navButtons={
      Array [
        Object {
          "icon": <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  496,
                  512,
                  Array [],
                  "f14e",
                  "M225.38 233.37c-12.5 12.5-12.5 32.76 0 45.25 12.49 12.5 32.76 12.5 45.25 0 12.5-12.5 12.5-32.76 0-45.25-12.5-12.49-32.76-12.49-45.25 0zM248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm126.14 148.05L308.17 300.4a31.938 31.938 0 0 1-15.77 15.77l-144.34 65.97c-16.65 7.61-33.81-9.55-26.2-26.2l65.98-144.35a31.938 31.938 0 0 1 15.77-15.77l144.34-65.97c16.65-7.6 33.8 9.55 26.19 26.2z",
                ],
                "iconName": "compass",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "label": "Explore",
          "path": "/explore",
        },
        Object {
          "icon": <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  384,
                  512,
                  Array [],
                  "f3c5",
                  "M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z",
                ],
                "iconName": "map-marker-alt",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "label": "Near Me",
          "path": "/nearme",
        },
        Object {
          "icon": <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f07a",
                  "M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z",
                ],
                "iconName": "shopping-cart",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "label": "My Cart",
          "path": "/mycart",
        },
        Object {
          "icon": <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f007",
                  "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                ],
                "iconName": "user",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "label": "Profile",
          "path": "/profile",
        },
      ]
    }
  />
</div>
`;
